### ISSUE TITLE
Error during installation in Angular 17

### ISSUE THREAD

**Issue by luanpaivadev** [sweetalert2/ngx-sweetalert2#261]
When trying to install the project using Angular 17, this error is occurring:

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: control-vendas-app@0.0.0
npm ERR! Found: @angular/common@17.0.9
npm ERR! node_modules/@angular/common
npm ERR! @angular/common@"^17.0.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/common@"^14.0.0 || ^15.0.0 || ^16.0.0" from @sweetalert2/ngx-sweetalert2@12.2.0
npm ERR! node_modules/@sweetalert2/ngx-sweetalert2
npm ERR! @sweetalert2/ngx-sweetalert2@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

Is there already a solution?

**Comment by luanpaivadev** [comment]
@danieljames08 

Downgrading is not an option, I am using features that only exist in version 17. :'(

**Comment by 10-ti** [comment]
npm install ... --legacy-peer-deps

**Comment by Lucasrsv1** [comment]
Until the PR [#252](https://github.com/sweetalert2/ngx-sweetalert2/pull/252) is merged, we can workaround this error by adding the following override to `package.json`:

```json
"overrides": {
	"@sweetalert2/ngx-sweetalert2": {
		"@angular/common": "^14.0.0 || ^15.0.0 || ^16.0.0 || ^17.0.0",
		"@angular/core": "^14.0.0 || ^15.0.0 || ^16.0.0 || ^17.0.0"
	}
}
```

I hope this helps.

**Comment by luanpaivadev** [comment]
The installation went well, but when trying to use it, it gives this error:

auth.guard.ts:21 ERROR NullInjectorError: R3InjectorError(Standalone[_PropostaGanhaTableComponent])[_SweetAlert2LoaderService -> _SweetAlert2LoaderService -> _SweetAlert2LoaderService -> _SweetAlert2LoaderService]:
   NullInjectorError: No provider for _SweetAlert2LoaderService!

I tried to configure the provider, but I couldn't... :(

**Comment by luanpaivadev** [comment]
When trying to place SweetAlert2LoaderService in appConfig, the error changes to this:

ERROR NullInjectorError: R3InjectorError(Standalone[_PropostaGanhaTableComponent])[_SweetAlert2LoaderService -> _SweetAlert2LoaderService -> _SweetAlert2LoaderService -> InjectionToken @sweetalert2/ngx-sweetalert2#swalProvider -> InjectionToken @sweetalert2/ngx-sweetalert2#swalProvid er]:
   NullInjectorError: No provider for InjectionToken @sweetalert2/ngx-sweetalert2#swalProvider!

**Comment by limonte** [comment]
:tada: This issue is solved in version 12.3.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@sweetalert2/ngx-sweetalert2/v/12.3.0)
- [GitHub release](https://github.com/sweetalert2/ngx-sweetalert2/releases/tag/v12.3.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
